File: funcscript/funcs/logic/NotEqualsFunction.cs
using funcscript.core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace funcscript.funcs.logic
{

    public class NotEqualsFunction : IFsFunction
    {
        public int MaxParsCount => 2;

        public CallType CallType => CallType.Infix;

        public string Symbol => "!=";

        public int Precidence => 200;

        public object Evaluate(IFsDataProvider parent, IParameterList pars)
        {
            if (pars.Count != this.MaxParsCount)
                throw new error.EvaluationTimeException($"{this.Symbol} function: invalid parameter count. {this.MaxParsCount} expected got {pars.Count}");
            var par0=pars[0];
            var par1 = pars[1];

            if (par0 == null && par1 == null)
                return false;

            if (par0 == null || par1 == null)
                return true;

            if (FuncScript.IsNumeric(par0) && FuncScript.IsNumeric(par1))
            {
                FuncScript.ConvertToCommonNumbericType(par0, par1,out par0,out par1);
            }

            if (par0.GetType() != par1.GetType())
                return true;
            
            return !par0.Equals(par1);

        }

        public string ParName(int index)
        {
            switch(index)
            {
                case 0:
                    return "Left Value";
                case 1:
                    return "Right Value";
                default:
                    return "";
            }
        }
    }
}

File: js_port/fs_js/src/funcs/logic/NotEqualsFunction.js
const { BaseFunction, CallType } = require('../../core/functionBase');
const helpers = require('../helpers');

class NotEqualsFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = '!=';
    this.callType = CallType.Infix;
    this.precidence = 200;
  }

  get maxParameters() {
    return 2;
  }

  evaluate(provider, parameters) {
    const error = helpers.expectParamCount(this.symbol, parameters, this.maxParameters);
    if (error) {
      return error;
    }
    let left = parameters.getParameter(provider, 0);
    let right = parameters.getParameter(provider, 1);

    if (helpers.typeOf(left) === helpers.FSDataType.Null && helpers.typeOf(right) === helpers.FSDataType.Null) {
      return helpers.makeValue(helpers.FSDataType.Boolean, false);
    }
    if (helpers.typeOf(left) === helpers.FSDataType.Null || helpers.typeOf(right) === helpers.FSDataType.Null) {
      return helpers.makeValue(helpers.FSDataType.Boolean, true);
    }

    if (helpers.isNumeric(left) && helpers.isNumeric(right)) {
      [left, right] = helpers.convertToCommonNumericType(left, right);
    }

    if (helpers.typeOf(left) !== helpers.typeOf(right)) {
      return helpers.makeValue(helpers.FSDataType.Boolean, true);
    }

    return helpers.makeValue(helpers.FSDataType.Boolean, helpers.valueOf(left) !== helpers.valueOf(right));
  }
}

module.exports = {
  NotEqualsFunction
};
